SPECS/numpy/numpy.spec

```
@@ -1,131 +1,857 @@
-
-%package -n     python3-numpy
-Summary:        python-numpy
-Requires:       python3
-
 %description -n python3-numpy
-Python 3 version.
+NumPy is a general-purpose array-processing package designed to
+efficiently manipulate large multi-dimensional arrays of arbitrary
+records without sacrificing too much speed for small multi-dimensional
+arrays.  NumPy is built on the Numeric code base and adds features
+introduced by numarray as well as an extended C-API and the ability to
+create arrays of arbitrary type.
+
+There are also basic facilities for discrete fourier transform,
+basic linear algebra and random number generation. Also included in
+this package is a version of f2py that works properly with NumPy.
 
 %package -n python3-numpy-f2py
+%{?python_provide:%python_provide python3-numpy-f2py}
 Summary:        f2py for numpy
+License:        BSD AND Python AND ASL 2.0
 Requires:       python3-devel
-Requires:       python3-numpy = %{version}-%{release}
+Requires:       python3-numpy%{?_isa} = %{version}-%{release}
 Provides:       python3-f2py = %{version}-%{release}
+Obsoletes:      python3-f2py <= 2.45.241_1927
+Provides:       f2py = %{version}-%{release}
+Provides:       numpy-f2py = %{version}-%{release}
+Obsoletes:      numpy-f2py < 1:1.10.1-3
 
 %description -n python3-numpy-f2py
 This package includes a version of f2py that works properly with NumPy.
 
+%package -n python3-numpy-doc
+Summary:        Documentation for numpy
+License:        BSD AND Python AND ASL 2.0
+Requires:       python3-numpy = %{version}-%{release}
+BuildArch:      noarch
+
+%description -n python3-numpy-doc
+This package provides the complete documentation for NumPy.
+
 %prep
 %autosetup -p1
 
+# Force re-cythonization (ifed for PKG-INFO presence in setup.py)
+# rm PKG-INFO
+
+# openblas is provided by flexiblas by default; otherwise,
+# Use openblas pthreads as recommended by upstream (see comment in site.cfg.example)
+cat >> site.cfg <<EOF
+[openblas]
+libraries = %{blaslib}%{blasvar}
+library_dirs = %{_libdir}
+EOF
+
 %build
-%py3_build
+%set_build_flags
+
+env OPENBLAS=%{_libdir} \
+    BLAS=%{_libdir} \
+    LAPACK=%{_libdir} CFLAGS="%{optflags}" \
+    %{__python3} setup.py build
 
-# using py3_install macro, numpy header files are missing. --skip-build arg is removed essentially with this change.
-# https://github.com/numpy/numpy/issues/16005
-# changes imported from Fedora.
 %install
-%{__python3} setup.py install --root %{buildroot}
+mkdir docs
+pushd docs
+unzip %{SOURCE1}
+popd
+
+#%%{__python3} setup.py install -O1 --skip-build --root %%{buildroot}
+# skip-build currently broken, this works around it for now
+env OPENBLAS=%{_libdir} \
+    FFTW=%{_libdir} BLAS=%{_libdir} \
+    LAPACK=%{_libdir} CFLAGS="%{optflags}" \
+    %{__python3} setup.py install --root %{buildroot} --prefix=%{_prefix}
+pushd %{buildroot}%{_bindir} &> /dev/null
+ln -s f2py3 f2py.numpy
+popd &> /dev/null
```

Nit-pick: this pushd-popd seems a bit excessive for what we're doing.
```suggestion
ln -s f2py3 %{buildroot}%{_bindir}/f2py.numpy
```